plugins/woocommerce-blocks/tests/e2e/tests/checkout/order-confirmation.block_theme.side_effects.spec.ts

```diff
@@ -109,9 +114,20 @@ test.describe( 'Shopper → Order Confirmation (logged in user)', () => {
 		// Confirm order details are not visible
 		await pageObject.verifyOrderConfirmationDetails( page, false );
 
-		// The following tests are skipped until the multiple sign in roles is implemented
-		// - Confirm details are hidden when logged out
-		// - Confirm data is hidden without valid session/key
+		await test.step( 'Logout the user and revisit the order received page to verify that details are displayed when woocommerce_order_received_verify_known_shoppers is disabled', async () => {
+			await requestUtils.activatePlugin(
+				'woocommerce-blocks-test-order-confirmation-filters'
+			);
+			await page.goto( '/my-account' );
+			await page
+				.locator(
+					'li.woocommerce-MyAccount-navigation-link--customer-logout a'
+				)
+				.click();
+			await expect( page ).toHaveURL( '/my-account' );
```

Comment:
I wasn't aware logging out doesn't redirect, so this bit doesn't really make sense – It will pass regardless of whether we log out successfully or not. 😅 Let's wait for the `Log in` button instead, as it's definitely something that doesn't exist on the logged-in `My Account` page:

```suggestion
			await page.goto( '/my-account' );
			await page
				.locator(
					'li.woocommerce-MyAccount-navigation-link--customer-logout a'
				)
				.click();
			await expect(
				page.getByRole( 'button', { name: 'Log in' } )
			).toBeVisible();
```

Sorry for the confusion!

Reply:
Ahah same, I did it and also didn't notice the URL was exactly the same although it was right there in my face.
Cool I also agree that checking for content might be a best bet 👍🏼 

Done in https://github.com/woocommerce/woocommerce/pull/46957/commits/e885f1d352cbfe21049780a67a62d66e2fb4fb42